pkg/helpers/sa_syncer.go

```diff
@@ -0,0 +1,67 @@
+package helpers
+
+import (
+	"context"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+)
+
+// EnsureKubeocnfigFromSA create a secret of kubeconfig with a service account user.
+// It will first get the SA token from the hosted cluster using hostedClient
+// Then it will create a kubeconfig secret on currentClient using currentClient
+// The secret name of kubeconfig should be saName+"-kubeconfig", and the namespace should be saNamespace
+func EnsureKubeconfigFromSA(ctx context.Context, saName, saNamespace string, templateKubeconfig *rest.Config, renderSAToken func([]byte) error, hostedClient kubernetes.Interface) error {
```

Comment:
I think it is better to not have templateKubeconfig as the parameter, which could be done in reader func

Reply:
A little confused here, if we move templateKubeconfig out of parameters, should this method still be called "EnsureKubeconfigFromSA" or just "EnsureSA"? Because users can render it in many ways not just kubeconfig.

Reply:
How about providing two functions for this?

The first function is:

```
func GetSAToken(sa, sanamespace string, hostedClient kubernetes.Interface) ([]byte, error) {...}
``` 

The second function is:
```
RenderKubeconfig(server, caData, saToken []byte) ([]byte, error) {...}
```
And the client code is:

```
...
saToken,err := helpers.GetSAToken(sa, saNamespace, hostedClient)
if err != nil {
  return err
}
saKubeconfig, err := helpers.RenderKubeconfig(adminKubeconfig.Host, adminKubeconfig. CAData, saToken)
if err != nil {
  return err
}
resourceApply.ApplyResources(...) // apply a secret object
...
```